react/features/base/color-scheme/ColorSchemeRegistry.js

```diff
@@ -0,0 +1,146 @@
+// @flow
+
+import { toState } from '../redux';
+import { StyleType } from '../styles';
+
+import defaultScheme from './defaultScheme';
+
+/**
+ * A registry class to register styles that need to be color-schemed.
+ *
+ * This class uses lazy initialization for scheme-ified style definitions on
+ * request.
+ */
+class ColorSchemeRegistry {
+    /**
+     * A map of registered style definitions.
+     */
+    _featureStyles = new Map();
```

Comment:
as dicussed offline: let's call this something with "template" in the name, since that's what it is